### PR TITLE
homework_09 complete

### PR DESCRIPTION
**Pull request by NikitaKuzminn**

### CODE REVIEW

File: src/main/java/home/epam/hw9/entities/BoardDto.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@NoArgsConstructor
+@Getter
+@Setter
```
Comment (kristinaChander) [inline src/main/java/home/epam/hw9/entities/BoardDto.java:11]: Why don't you use @Data

File: src/main/java/home/epam/hw9/entities/BoardDto.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+public class BoardDto {
+
+    @SerializedName("id")
```
Comment (kristinaChander) [inline src/main/java/home/epam/hw9/entities/BoardDto.java:14]: What does this annotation do? Do you need it?

File: src/main/java/home/epam/hw9/entities/BoardDto.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    @Expose
```
Comment (kristinaChander) [inline src/main/java/home/epam/hw9/entities/BoardDto.java:15]: What does this annotation do? Do you need it?

File: src/main/java/home/epam/hw9/entities/ListDto.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@NoArgsConstructor
+@Getter
+@Setter
```
Comment (kristinaChander) [inline src/main/java/home/epam/hw9/entities/ListDto.java:11]: @Data?

File: src/main/java/home/epam/hw9/entities/ListDto.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+public class ListDto {
+
+    @SerializedName("id")
+    @Expose
```
Comment (kristinaChander) [inline src/main/java/home/epam/hw9/entities/ListDto.java:15]: Annotations?

File: src/main/java/home/epam/hw9/service/BoardService.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import java.util.Map;
+
+public class BoardService extends CommonService {
+    public static final URI BOARD_URL = URI.create(URL_BOARD);
+
+    public static BoardApiBuilder apiBoardBuilder() {
+        return new BoardApiBuilder();
+    }
+
+    public static class BoardApiBuilder {
```
Comment (kristinaChander) [inline src/main/java/home/epam/hw9/service/BoardService.java:23]: Naming convention - class name should be like "BoardServiceBuilder"

File: src/main/java/home/epam/hw9/service/CommonService.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+    public String stringConverter() {
+        StringBuilder builder = new StringBuilder();
+        pathParameters.keySet().forEach(p -> builder.append("{").append(p).append("}").append("/"));
+        return builder.toString();
+    }
+
+    public Response sendRequest(RequestSpecification request) {
+        return RestAssured
+            .given(request).log().all()
```
Comment (kristinaChander) [inline src/main/java/home/epam/hw9/service/CommonService.java:56]: do you need all the logs?

File: src/main/java/home/epam/hw9/service/CommonService.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            .pathParams(pathParameters)
+            .queryParams(queryParameters)
+            .queryParam(KEY, USER_KEY)
+            .queryParam(TOKEN, USER_TOKEN)
+            .request(requestedMethod, stringConverter())
+            .prettyPeek();
+    }
+
+    public static RequestSpecification requestSpec(URI url) {
```
Comment (kristinaChander) [inline src/main/java/home/epam/hw9/service/CommonService.java:65]: emove static - don't use static for inheritance

File: src/main/java/home/epam/hw9/service/ListService.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.Map;
+
+public class ListService extends CommonService {
+    public static final URI LIST_URL = URI.create(LISTS_URL);
+
+    public static ListApiBuilder listApiBuilder() {
+        return new ListApiBuilder();
+    }
+
+    public static class ListApiBuilder {
```
Comment (kristinaChander) [inline src/main/java/home/epam/hw9/service/ListService.java:24]: change name

File: src/main/java/home/epam/hw9/util/ConfProperty.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+public class ConfProperty {
+
+    public static String getProperty(String name) {
+        FileInputStream fis;
+        String properties = null;
+        Properties property = new Properties();
+        try {
+            fis = new FileInputStream("src/test/resources/home.epam.hw9/test.properties");
+            property.load(fis);
+            properties = property.getProperty(name);
```
Comment (kristinaChander) [inline src/main/java/home/epam/hw9/util/ConfProperty.java:16]: Shall we close the stream?

File: src/main/java/home/epam/hw9/util/Values.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package home.epam.hw9.util;
+
+import static home.epam.hw9.util.ConfProperty.getProperty;
+
+public class Values {
+
+    public static final String USER_KEY = getProperty("key");
```
Comment (kristinaChander) [inline src/main/java/home/epam/hw9/util/Values.java:7]: you read from file every time you invoke getProperty() method, can you read once and then use the already read info?

File: src/test/java/home/epam/hw9/data/DataProviders.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+import static home.epam.hw9.util.Values.BOARD_NAME;
+import static home.epam.hw9.util.Values.LIST_NAME;
+
+import home.epam.hw9.entities.BoardDto;
+import home.epam.hw9.entities.ListDto;
+import org.testng.annotations.DataProvider;
+
+public class DataProviders {
+
+    @DataProvider
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/data/DataProviders.java:12]: you only use name, you don't need a DTO in your requests

File: src/test/java/home/epam/hw9/data/DataProviders.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    public static Object[][] boardData() {
+        BoardDto boardDto = new BoardDto();
+        boardDto.setName(BOARD_NAME);
+        return new Object[][] {
+            {boardDto}
+        };
+    }
+
+    @DataProvider
+    public static Object[][] listData() {
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/data/DataProviders.java:22]: you only use name, you don't need a DTO in your requests

File: src/test/java/home/epam/hw9/steps/BoardSteps.java [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+import home.epam.hw9.entities.BoardDto;
+import home.epam.hw9.service.BoardService;
+import io.qameta.allure.Step;
+import io.restassured.http.Method;
+import io.restassured.response.Response;
+import io.restassured.specification.ResponseSpecification;
+
+public class BoardSteps {
+
+    public Response sendRequestAndGetResponse(BoardService.BoardApiBuilder api, ResponseSpecification resp) {
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/steps/BoardSteps.java:18]: change method name

File: src/test/java/home/epam/hw9/steps/BoardSteps.java [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+            .sendRequest();
+        response.then()
+                .assertThat()
+                .spec(resp);
+        return response;
+    }
+
+    @Step("Create new board")
+    public String createNewBoard(BoardDto boardDto) {
+        BoardService.BoardApiBuilder api = apiBoardBuilder()
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/steps/BoardSteps.java:30]: change the variable name - it's a request - the name should reflect the main idea

File: src/test/java/home/epam/hw9/steps/BoardSteps.java [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+import home.epam.hw9.service.BoardService;
+import io.qameta.allure.Step;
+import io.restassured.http.Method;
+import io.restassured.response.Response;
+import io.restassured.specification.ResponseSpecification;
+
+public class BoardSteps {
+
+    public Response sendRequestAndGetResponse(BoardService.BoardApiBuilder api, ResponseSpecification resp) {
+        Response response = api
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/steps/BoardSteps.java:19]: change the variable name - it's a request - the name should reflect the main idea

File: src/test/java/home/epam/hw9/steps/BoardSteps.java [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+            .buildBoardApiRequest()
+            .sendRequest();
+        response.then()
+                .assertThat()
+                .spec(resp);
+        return response;
+    }
+
+    @Step("Create new board")
+    public String createNewBoard(BoardDto boardDto) {
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/steps/BoardSteps.java:29]: you don't need DTO - only name

File: src/test/java/home/epam/hw9/steps/BoardSteps.java [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+        BoardService.BoardApiBuilder api = apiBoardBuilder()
+            .setMethod(Method.POST)
+            .setName(boardDto.getName());
+        return extractBoardFromJson(sendRequestAndGetResponse(api, okResponseSpec())).getId();
+    }
+
+    @Step("Get board by id")
+    public BoardDto getBoard(String boardId) {
+        BoardService.BoardApiBuilder api = apiBoardBuilder()
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/steps/BoardSteps.java:38]: variable name

File: src/test/java/home/epam/hw9/steps/BoardSteps.java [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+            .setMethod(Method.GET)
+            .setId(boardId);
+        return extractBoardFromJson(sendRequestAndGetResponse(api, okResponseSpec()));
+    }
+
+    @Step("Delete board by id")
+    public void deleteBoard(String boardId) {
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/steps/BoardSteps.java:45]: variable name

File: src/test/java/home/epam/hw9/steps/BoardSteps.java [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+    public Response getDeletedBoard(String boardId) {
+        BoardService.BoardApiBuilder builder = apiBoardBuilder()
+            .setMethod(Method.GET)
+            .setId(boardId);
+        return sendRequestAndGetResponse(builder, notFoundResponseSpec());
+    }
+
+    @Step("Update board name by id")
+    public BoardDto updateBoardName(String boardId) {
+        BoardService.BoardApiBuilder api = apiBoardBuilder()
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/steps/BoardSteps.java:62]: variable name

File: src/test/java/home/epam/hw9/steps/ListSteps.java [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+import home.epam.hw9.entities.ListDto;
+import home.epam.hw9.service.ListService;
+import io.qameta.allure.Step;
+import io.restassured.http.Method;
+import io.restassured.response.Response;
+import io.restassured.specification.ResponseSpecification;
+
+public class ListSteps {
+
+    public ListDto sendRequestAndGetResponse(ListService.ListApiBuilder api, ResponseSpecification resp) {
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/steps/ListSteps.java:17]: change method name

File: src/test/java/home/epam/hw9/steps/ListSteps.java [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+        Response response = api
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/steps/ListSteps.java:18]: variable name

File: src/test/java/home/epam/hw9/steps/ListSteps.java [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+            .buildListApiRequest()
+            .sendRequest();
+        response.then()
+                .assertThat()
+                .spec(resp);
+        return extractListFromJson(response);
+    }
+
+    @Step("Create new list")
+    public String createList(String boardId, ListDto listDto) {
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/steps/ListSteps.java:28]: you don't need dto

File: src/test/java/home/epam/hw9/tests/BoardDtoTest.java [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+
+import home.epam.hw9.data.DataProviders;
+import home.epam.hw9.entities.BoardDto;
+import io.restassured.response.Response;
+import org.testng.annotations.Test;
+
+public class BoardDtoTest extends BaseTest {
+
+    @Test(dataProvider = "boardData", dataProviderClass = DataProviders.class)
+    public void createBoardTest(BoardDto boardDto) {
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/tests/BoardDtoTest.java:18]: you don't need DTO - only name

File: src/test/java/home/epam/hw9/tests/BoardDtoTest.java [hunk 25]
```diff
@@ -0,0 +1,40 @@
...
+        boardId = boardSteps.createNewBoard(boardDto);
+        BoardDto createdBoardDto = boardSteps.getBoard(boardId);
+        assertThat(createdBoardDto.getName(), equalTo(boardDto.getName()));
+    }
+
+    @Test(dataProvider = "boardData", dataProviderClass = DataProviders.class)
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/tests/BoardDtoTest.java:24]: you don't need DTO - only name

File: src/test/java/home/epam/hw9/tests/BoardDtoTest.java [hunk 26]
```diff
@@ -0,0 +1,40 @@
...
+    public void deleteBoardTest(BoardDto boardDto) {
+        boardId = boardSteps.createNewBoard(boardDto);
+        boardSteps.deleteBoard(boardId);
+        Response response = boardSteps.getDeletedBoard(boardId);
+        assertThat(response.getStatusCode(), equalTo(NOT_FOUND_STATUS));
+        assertThat(response.getStatusLine(), containsStringIgnoringCase(NOT_FOUND_MESSAGE));
+        boardId = null;
+    }
+
+    @Test(dataProvider = "boardData", dataProviderClass = DataProviders.class)
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/tests/BoardDtoTest.java:34]: you don't need DTO - only name

File: src/test/java/home/epam/hw9/tests/ListDtoTest.java [hunk 27]
```diff
@@ -0,0 +1,32 @@
...
+
+import home.epam.hw9.data.DataProviders;
+import home.epam.hw9.entities.BoardDto;
+import home.epam.hw9.entities.ListDto;
+import home.epam.hw9.steps.ListSteps;
+import org.testng.annotations.Test;
+
+public class ListDtoTest extends BaseTest {
+
+    @Test(dataProvider = "listData", dataProviderClass = DataProviders.class)
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/tests/ListDtoTest.java:15]: you don't need DTO - only name

File: src/test/java/home/epam/hw9/tests/ListDtoTest.java [hunk 28]
```diff
@@ -0,0 +1,32 @@
...
+    public void createListTest(BoardDto boardDto, ListDto listDto) {
+        listSteps = new ListSteps();
+        boardId = boardSteps.createNewBoard(boardDto);
+        listId = listSteps.createList(boardId, listDto);
+        ListDto createdListDto = listSteps.getList(listId);
+        assertThat(createdListDto.getName(), equalTo(listDto.getName()));
+    }
+
+    @Test(dataProvider = "listData", dataProviderClass = DataProviders.class)
+    public void modifyListTest(BoardDto boardDto, ListDto listDto) {
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/tests/ListDtoTest.java:25]: you don't need DTO - only name

File: src/test/java/home/epam/hw9/tests/ListDtoTest.java [hunk 29]
```diff
@@ -0,0 +1,32 @@
...
+import home.epam.hw9.entities.BoardDto;
+import home.epam.hw9.entities.ListDto;
+import home.epam.hw9.steps.ListSteps;
+import org.testng.annotations.Test;
+
+public class ListDtoTest extends BaseTest {
+
+    @Test(dataProvider = "listData", dataProviderClass = DataProviders.class)
+    public void createListTest(BoardDto boardDto, ListDto listDto) {
+        listSteps = new ListSteps();
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/tests/ListDtoTest.java:17]: code duplication

File: src/test/java/home/epam/hw9/tests/ListDtoTest.java [hunk 30]
```diff
@@ -0,0 +1,32 @@
...
+        boardId = boardSteps.createNewBoard(boardDto);
+        listId = listSteps.createList(boardId, listDto);
+        ListDto createdListDto = listSteps.getList(listId);
+        assertThat(createdListDto.getName(), equalTo(listDto.getName()));
+    }
+
+    @Test(dataProvider = "listData", dataProviderClass = DataProviders.class)
+    public void modifyListTest(BoardDto boardDto, ListDto listDto) {
+        listSteps = new ListSteps();
```
Comment (kristinaChander) [inline src/test/java/home/epam/hw9/tests/ListDtoTest.java:26]: code duplication listSteps = new ListSteps();

**Review by kristinaChander — CHANGES_REQUESTED** [review]
Please, resolve the comments

File: src/main/java/home/epam/hw9/entities/BoardDto.java (hunk 1 above)
Comment (NikitaKuzminn) [reply src/main/java/home/epam/hw9/entities/BoardDto.java:11]: fixed

File: src/main/java/home/epam/hw9/entities/BoardDto.java (hunk 2 above)
Comment (NikitaKuzminn) [reply src/main/java/home/epam/hw9/entities/BoardDto.java:14]: fixed

File: src/main/java/home/epam/hw9/entities/BoardDto.java (hunk 3 above)
Comment (NikitaKuzminn) [reply src/main/java/home/epam/hw9/entities/BoardDto.java:15]: fixed

File: src/main/java/home/epam/hw9/entities/ListDto.java (hunk 4 above)
Comment (NikitaKuzminn) [reply src/main/java/home/epam/hw9/entities/ListDto.java:11]: fixed

File: src/main/java/home/epam/hw9/entities/ListDto.java (hunk 5 above)
Comment (NikitaKuzminn) [reply src/main/java/home/epam/hw9/entities/ListDto.java:15]: fixed

File: src/main/java/home/epam/hw9/service/BoardService.java (hunk 6 above)
Comment (NikitaKuzminn) [reply src/main/java/home/epam/hw9/service/BoardService.java:23]: fixed

File: src/main/java/home/epam/hw9/service/CommonService.java (hunk 7 above)
Comment (NikitaKuzminn) [reply src/main/java/home/epam/hw9/service/CommonService.java:56]: fixed

File: src/main/java/home/epam/hw9/service/CommonService.java (hunk 8 above)
Comment (NikitaKuzminn) [reply src/main/java/home/epam/hw9/service/CommonService.java:65]: fixed

File: src/main/java/home/epam/hw9/service/ListService.java (hunk 9 above)
Comment (NikitaKuzminn) [reply src/main/java/home/epam/hw9/service/ListService.java:24]: fixed

File: src/main/java/home/epam/hw9/util/ConfProperty.java (hunk 10 above)
Comment (NikitaKuzminn) [reply src/main/java/home/epam/hw9/util/ConfProperty.java:16]: fixed

File: src/test/java/home/epam/hw9/data/DataProviders.java (hunk 12 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/data/DataProviders.java:12]: fixed

File: src/test/java/home/epam/hw9/data/DataProviders.java (hunk 13 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/data/DataProviders.java:22]: fixed

File: src/test/java/home/epam/hw9/steps/BoardSteps.java (hunk 15 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/steps/BoardSteps.java:30]: fixed

File: src/test/java/home/epam/hw9/steps/BoardSteps.java (hunk 16 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/steps/BoardSteps.java:19]: fixed

File: src/test/java/home/epam/hw9/steps/BoardSteps.java (hunk 17 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/steps/BoardSteps.java:29]: fixed

File: src/test/java/home/epam/hw9/steps/BoardSteps.java (hunk 18 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/steps/BoardSteps.java:38]: fixed

File: src/test/java/home/epam/hw9/steps/BoardSteps.java (hunk 19 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/steps/BoardSteps.java:45]: fixed

File: src/test/java/home/epam/hw9/steps/BoardSteps.java (hunk 20 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/steps/BoardSteps.java:62]: fixed

File: src/test/java/home/epam/hw9/steps/ListSteps.java (hunk 22 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/steps/ListSteps.java:18]: fixed

File: src/test/java/home/epam/hw9/steps/ListSteps.java (hunk 23 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/steps/ListSteps.java:28]: fixed

File: src/test/java/home/epam/hw9/tests/BoardDtoTest.java (hunk 24 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/tests/BoardDtoTest.java:18]: fixed

File: src/test/java/home/epam/hw9/tests/BoardDtoTest.java (hunk 25 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/tests/BoardDtoTest.java:24]: fixed

File: src/test/java/home/epam/hw9/tests/BoardDtoTest.java (hunk 26 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/tests/BoardDtoTest.java:34]: fixed

File: src/test/java/home/epam/hw9/tests/ListDtoTest.java (hunk 27 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/tests/ListDtoTest.java:15]: fixed

File: src/test/java/home/epam/hw9/tests/ListDtoTest.java (hunk 28 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/tests/ListDtoTest.java:25]: fixed

File: src/test/java/home/epam/hw9/tests/ListDtoTest.java (hunk 29 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/tests/ListDtoTest.java:17]: fixed

File: src/test/java/home/epam/hw9/tests/ListDtoTest.java (hunk 30 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/tests/ListDtoTest.java:26]: fixed

File: src/test/java/home/epam/hw9/steps/BoardSteps.java (hunk 14 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/steps/BoardSteps.java:18]: fixed

File: src/test/java/home/epam/hw9/steps/ListSteps.java (hunk 21 above)
Comment (NikitaKuzminn) [reply src/test/java/home/epam/hw9/steps/ListSteps.java:17]: fixed

File: src/main/java/home/epam/hw9/util/Values.java (hunk 11 above)
Comment (NikitaKuzminn) [reply src/main/java/home/epam/hw9/util/Values.java:7]: fixed

**Pull request merged**